adapt/context.py

```diff
@@ -8,18 +8,26 @@ class ContextManagerFrame(object):
     Manages entities and context for a single frame of conversation.
     Provides simple equality querying.
     """
-    def __init__(self, entities=[], metadata={}):
+    def __init__(self, entities=None, metadata=None):
         self.entities = entities
         self.metadata = metadata
 
-    def metadata_matches(self, query={}):
+    def metadata_matches(self, query=None):
+        if self.metadata is None:
+            self.metadata = {}
+        if query is None:
+            query = {}
         result = len(query.keys()) > 0
         for key in query.keys():
             result = result and query[key] == self.metadata.get(key)
 
         return result
 
     def merge_context(self, tag, metadata):
+        if metadata is None:
+            metadata = {}
+        if self.entities is None:
+            self.entities = []
```

Comment:
And this self.entities check isn't needed if you implement the above constructor change suggested above.